src/js/module.js

```diff
@@ -174,6 +175,9 @@ iotjs_module_t.load = function(id, parent) {
   if (process.builtin_modules[id]) {
     return Native.require(id);
   }
+  if (iotjs_module_t.remoteCache[id]) {
+    return iotjs_module_t.remoteCache[id].exports;
+  }
```

Comment:
@DanielBallaSZTE `loadRemote` seems be for remote loading, and `load` is for both of local and remote. I think all logic related to remote loading could be placed in `loadRemote`. Is it possible?

Reply:
Yes, merging the two functions is definitely possible.
On the one hand the source loading would be handled within one function, which would be nice indeed.
On the other hand though, the function would require a couple more parameters, and several if-else cases, which -in my opinion- wouldn't look that great.

The lines you highlighted are not really for loading both remote and local modules.
Remote modules are handled only in `loadRemote`, the only thing it checks, is whether the module that would be loaded locally is already present in the remoteCache.

Reply:
@DanielBallaSZTE Why do we need to check the remoteCache exists before loading local module?

Reply:
@glistening Oh, you are right. Local modules are loaded before remotes would be, therefore there's no need to check whether a remote modules exists with the same name. Thanks for that! Somehow I missed that part.